BPOR/NIHR.NotificationService/Services/NotificationService.cs

```diff
@@ -99,70 +127,139 @@ private async Task SendBatchWithRateLimitAsync(List<Notification> batch,
+            await retryPolicy.ExecuteAsync(async () => { await SendNotificationAsync(sendNotificationRequest, cancellationToken); });
             stopwatch.Stop();
+
             lock (individualTimes)
             {
                 individualTimes.Add(stopwatch.ElapsedMilliseconds);
             }
         }
 
-        private static SendEmailRequest CreateSendEmailRequest(string email, Dictionary<string, string> personalisation)
+        private static SendNotificationRequest CreateSendNotificationRequest(Dictionary<string, string> personalisation)
         {
-            if (!personalisation.TryGetValue("emailCampaignParticipantId", out var reference))
+            if (!personalisation.TryGetValue("campaignParticipantId", out var reference))
             {
-                throw new KeyNotFoundException($"EmailCampaignParticipantId not found for email: {email}");
+                throw new KeyNotFoundException("campaignParticipantId not found in personalisation data.");
             }
-            
-            if (!personalisation.TryGetValue("emailTemplateId", out var emailTemplateId))
+
+            if (!personalisation.TryGetValue("templateId", out var templateId))
+            {
+                throw new KeyNotFoundException("templateId not found in personalisation data.");
+            }
+
+            if (!personalisation.TryGetValue("campaignTypeId", out var campaignTypeIdStr))
             {
-                throw new KeyNotFoundException($"EmailTemplateId not found for email: {email}");
+                throw new KeyNotFoundException("campaignTypeId not found in personalisation data.");
             }
 
-            return new SendEmailRequest
+            if (!int.TryParse(campaignTypeIdStr, out var campaignTypeId))
             {
-                EmailAddress = email,
-                EmailTemplateId = emailTemplateId,
+                throw new ArgumentException($"campaignTypeId '{campaignTypeIdStr}' is not a valid integer.");
+            }
+
+            var request = new SendNotificationRequest
+            {
+                TemplateId = templateId,
                 Personalisation = personalisation,
-                Reference = reference
+                Reference = reference,
             };
+
+            var contactMethod = (ContactMethod)campaignTypeId;
+
+            switch (contactMethod)
+            {
+                case ContactMethod.Email:
+                    if (!personalisation.TryGetValue("email", out var email))
+                    {
+                        throw new KeyNotFoundException("Email address not found for email notification.");
+                    }
+                    request.EmailAddress = email;
+                    break;
+
+                case ContactMethod.Letter:
+                    if (!personalisation.TryGetValue("address_line_1", out var addressLine1) ||
+                        !personalisation.TryGetValue("address_line_2", out var addressLine2) ||
```

Comment:
Address line 2 is not a required field for sending letters normally, if this is a notify restriction it will have a knock on effect on both VS and the new RMS offline journey where address line 2 is optional.